core/modules/widgets/checkbox.js

```diff
@@ -3,129 +3,134 @@ title: $:/core/modules/widgets/checkbox.js
 	}
-	return false;
+	return(value === this.checkboxChecked) ? true
+	: (value === this.checkboxUnchecked) ? false
+	: (this.checkboxInvertTag) ? true : false;
 };
 
+/*
+Invoke the action associated with this widget
+*/
 CheckboxWidget.prototype.handleChangeEvent = function(event) {
-	var checked = this.inputDomNode.checked,
-		tiddler = this.wiki.getTiddler(this.checkboxTitle),
-		fallbackFields = {text: ""},
-		newFields = {title: this.checkboxTitle},
+	var tiddler = this.wiki.getTiddler(this.checkboxTitle),
+		fallbackFields = {
+			text: ""
+		},
+		newFields = {
+			title: this.checkboxTitle,
+		},
+		hastag = (tiddler && tiddler.hasTag(this.checkboxTag)),
 		hasChanged = false,
-		tagCheck = false,
-		hasTag = tiddler && tiddler.hasTag(this.checkboxTag);
-	if(this.checkboxTag && this.checkboxInvertTag === "yes") {
-		tagCheck = hasTag === checked;
-	} else {
-		tagCheck = hasTag !== checked;
-	}
+		data,
+		inverted = this.checkboxInvertTag === "yes",
+		checked = this.inputDomNode.checked,
+		value = checked ? this.checkboxChecked : this.checkboxUnchecked,
+		tagCheck = (this.checkboxTag && inverted) ? hastag === checked
+		: hastag !== checked;
 	// Set the tag if specified
 	if(this.checkboxTag && (!tiddler || tagCheck)) {
 		newFields.tags = tiddler ? (tiddler.fields.tags || []).slice(0) : [];
 		var pos = newFields.tags.indexOf(this.checkboxTag);
 		if(pos !== -1) {
-			newFields.tags.splice(pos,1);
+			newFields.tags.splice(pos, 1);
 		}
-		if(this.checkboxInvertTag === "yes" && !checked) {
-			newFields.tags.push(this.checkboxTag);
-		} else if(this.checkboxInvertTag !== "yes" && checked) {
+		if(inverted) {
+			if(!checked) {
+				newFields.tags.push(this.checkboxTag);
+			}
+		} else if(checked) {
 			newFields.tags.push(this.checkboxTag);
 		}
 		hasChanged = true;
 	}
 	// Set the field if specified
 	if(this.checkboxField) {
-		var value = checked ? this.checkboxChecked : this.checkboxUnchecked;
-		if(!tiddler || tiddler.fields[this.checkboxField] !== value) {
+		data = (tiddler) ? tiddler.fields[this.checkboxField] : "";
+		if(!tiddler || data !== value) {
 			newFields[this.checkboxField] = value;
 			hasChanged = true;
 		}
+	// Set the index if specified
+	} else if(this.checkboxIndex) {
+		data = this.wiki.getTiddlerData(this.checkboxTitle, {});
+		if(!tiddler || data[this.checkboxIndex] !== value) {
+			data[this.checkboxIndex] = value;
+			hasChanged = true;
+		}
 	}
+	// Update the tiddler if value has changed
 	if(hasChanged) {
-		this.wiki.addTiddler(new $tw.Tiddler(this.wiki.getCreationFields(),fallbackFields,tiddler,newFields,this.wiki.getModificationFields()));
+		if(this.checkboxField || this.checkboxTag) {
+			this.wiki.addTiddler(new $tw.Tiddler(this.wiki.getCreationFields(),
+				fallbackFields, tiddler, newFields, this.wiki.getModificationFields()));
+		} else if(this.checkboxIndex) {
+			this.wiki.setTiddlerData(this.checkboxTitle, data, this.wiki.getModificationFields(),
```

Comment:
@tobibeer 
It would help if there was better documentation for the functions in `utils.js` and `wiki.js`:-/

Looking at the function again, it would appear that modificationFields and creationFields are already catered for within the `setTiddlerData()` function -- I only need to pass the title and the data  (which is why the code worked -- the last two parameters were redundant.)